Move the CUDA implementation of ceil to ATen.

xuhdev opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #25372 Move the CUDA implementation of floor to ATen.
* #25285 Move the CUDA implementation of rsqrt to ATen.
* #25041 Move the CUDA implementation of round to ATen.
* #24879 Create helpers for implementing unary ops whose CUDA implementation is ATen.
* **#24866 Move the CUDA implementation of ceil to ATen.**

Fix #24542

Pull Request resolved: https://github.com/pytorch/pytorch/pull/24866

Differential Revision: [D16965903](https://our.internmc.facebook.com/intern/diff/D16965903)

xuhdev committed:
Move the CUDA implementation of ceil to ATen.

Fix #24542

xuhdev committed:
Update on "Move the CUDA implementation of ceil to ATen."


Fix #24542

xuhdev committed:
Update on "Move the CUDA implementation of ceil to ATen."


Fix #24542

xuhdev committed:
Update on "Move the CUDA implementation of ceil to ATen."


Fix #24542

xuhdev committed:
Update on "Move the CUDA implementation of ceil to ATen."


Fix #24542

xuhdev committed:
Update on "Move the CUDA implementation of ceil to ATen."


Fix #24542

xuhdev committed:
Update on "Move the CUDA implementation of ceil to ATen."


Fix #24542

xuhdev committed:
Update on "Move the CUDA implementation of ceil to ATen."


Fix #24542

xuhdev committed:
Update on "Move the CUDA implementation of ceil to ATen."


Fix #24542

xuhdev committed:
Update on "Move the CUDA implementation of ceil to ATen."


Fix #24542

xuhdev committed:
Update on "Move the CUDA implementation of ceil to ATen."


Fix #24542

xuhdev committed:
Update on "Move the CUDA implementation of ceil to ATen."


Fix #24542

xuhdev commented:
@VitalyFedyunin This PR should be ready for review now

VitalyFedyunin commented on aten/src/ATen/native/cuda/UnaryOpsKernel.cu:
```diff
@@ -32,6 +32,14 @@ void logical_not_kernel_cuda(TensorIterator& iter) {
+void ceil_kernel_cuda(TensorIterator& iter) {
+  AT_DISPATCH_FLOATING_TYPES_AND_HALF(iter.dtype(), "ceil_cuda", [&]() {
+    gpu_kernel(iter, []GPU_LAMBDA(scalar_t a) -> scalar_t {
+      return std::ceil(a);
```
`std::ceil` operates with `double` we need to use `ceilf` for floats for performance reasons.

xuhdev replied:
I thought `std::ceil` is overloaded for different floating point types? See https://en.cppreference.com/w/cpp/numeric/math/ceil

VitalyFedyunin replied:
Ah, right, overloaded in C++11. Good point.

VitalyFedyunin reviewed (commented):
Please check usage `ceil` vs `ceilf`

xuhdev committed:
Update on "Move the CUDA implementation of ceil to ATen."


Fix #24542

Differential Revision: [D16965903](https://our.internmc.facebook.com/intern/diff/D16965903)

xuhdev committed:
Update on "Move the CUDA implementation of ceil to ATen."


Fix #24542

Pull Request resolved: https://github.com/pytorch/pytorch/pull/24866

Differential Revision: [D16965903](https://our.internmc.facebook.com/intern/diff/D16965903)

xuhdev committed:
Update on "Move the CUDA implementation of ceil to ATen."


Fix #24542

Pull Request resolved: https://github.com/pytorch/pytorch/pull/24866

Differential Revision: [D16965903](https://our.internmc.facebook.com/intern/diff/D16965903)

xuhdev committed:
Update on "Move the CUDA implementation of ceil to ATen."


Fix #24542

Pull Request resolved: https://github.com/pytorch/pytorch/pull/24866

Differential Revision: [D16965903](https://our.internmc.facebook.com/intern/diff/D16965903)

xuhdev committed:
Update on "Move the CUDA implementation of ceil to ATen."


Fix #24542

Pull Request resolved: https://github.com/pytorch/pytorch/pull/24866

Differential Revision: [D16965903](https://our.internmc.facebook.com/intern/diff/D16965903)
